controller/server.go

```diff
@@ -104,16 +105,67 @@ func (rw *ReadWrite) TaskInspect(ctx context.Context, taskConfig config.TaskConf
 	return plan.ChangesPresent, plan.Plan, "", err
 }
 
-func (rw *ReadWrite) TaskRun(ctx context.Context, taskName string) error {
+func (rw *ReadWrite) TaskUpdate(ctx context.Context, taskConfig config.TaskConfig, runOp string) (bool, string, string, error) {
+	// Only enabled changes are supported at this time
+	if taskConfig.Enabled == nil {
+		return false, "", "", nil
+	}
+
+	if taskConfig.Name == nil || *taskConfig.Name == "" {
+		return false, "", "", fmt.Errorf("task name is required")
+	}
+
+	taskName := *taskConfig.Name
+	logger := rw.logger.With(taskNameLogKey, taskName)
+	logger.Trace("updating task")
+	if rw.drivers.IsActive(taskName) {
+		return false, "", "", fmt.Errorf("task '%s' is active and cannot be updated at this time", taskName)
+	}
+	rw.drivers.SetActive(taskName)
```

Comment:
This gets set to active as sort of a "non-blocking" mutex I guess?

Reply:
yeah the idea was to have a non-blocking mutex, mostly to preserve the original intention of delete behavior before the refactor.